src/components/Router.js

```diff
@@ -27,36 +27,46 @@ import EventNew from '../pages/admin/EventNew'
 import EventEdit from '../pages/admin/EventEdit'
 
 import { setUser } from '../actions/UserActions'
-import { log, getEvents } from '../utils'
+import {
+  log,
+  updateEvents,
+  updateUser
+} from '../utils'
 
 class Router extends Component {
   getAuthenticatedUser() {
     Auth.currentAuthenticatedUser({ bypassCache: true })
       .then(authUser => {
         const email = authUser.attributes.email
         if (email.substring(email.indexOf('@') + 1, email.length) === 'ubcbiztech.com') {
-          this.props.setUser({ ...authUser, admin: true });
+          this.props.setUser({
+            // name: authUser.attributes.name, // we don't need admin name for now
+            email: authUser.attributes.email,
+            admin: true
+          });
         }
         else {
-          console.log('not using a biztech e-mail!');
-          this.props.setUser({ ...authUser, admin: false });
+          const studentId = authUser.attributes['custom:student_id']
+          if (studentId) {
+            updateUser(studentId)
+          }
```

Comment:
If the user has not registered in the database, `studentId` will always be empty. This should do it I think..
```suggestion
          } else {
            this.props.setUser({
              name: authUser.attributes.name,
              email: authUser.attributes.email,
              admin: false
            });
          }
```

Reply:
That's not the case for admins, because the userpool could have a studentId it got from dev but no database entry on prod

Reply:
It only happens to non-biztech-email users, when they're not signed up in the DB yet

Reply:
sorry, I meant it can happen if we sign up with non-biztech emails on dev and then try to log in on prod (and vice versa)

Reply:
Yikes this is an interesting edge case

Reply:
facts